api/src/main/java/org/openmrs/module/clientregistry/api/ClientRegistryManager.java

```diff
@@ -82,4 +92,26 @@ public void disableClientRegistry() {
 		
 		isRunning.set(false);
 	}
+
+	/**
+	 * Determine the appropriate PatientService class based off of the client registry transaction type configuration
+	 * @return PatientService class corresponding to the appropriate transaction type supported by the client registry
+	 * @throws IllegalArgumentException if defined transaction type is unsupported
+	 */
+	public ClientRegistryPatientService getPatientService() throws IllegalArgumentException {
+		try {
+			String transactionMethodGlobalProperty = clientRegistryConfig.getClientRegistryTransactionMethod().toUpperCase();
+
+			switch (ClientRegistryTransactionType.valueOf(transactionMethodGlobalProperty)) {
+				case FHIR:
+					return fhirPatientService;
+				case HL7:
+					throw new IllegalArgumentException("HL7 transaction type is currently unsupported");
+			}
+		} catch (Exception ignored) {
+
+		}
+
+		throw new IllegalArgumentException("Unsupported transaction type");
```

Comment:
i should be able to put this in the catch statement above, but i don't think the compiler understands that all the code paths will either return or throw an exception...unless i'm looking at it weirdly